marker/builders/ocr.py

```diff
@@ -59,64 +48,61 @@ def get_recognition_batch_size(self):
             return 32
         return 32
 
-    def get_detection_batch_size(self):
-        if self.detection_batch_size is not None:
-            return self.detection_batch_size
-        elif settings.TORCH_DEVICE_MODEL == "cuda":
-            return 4
-        return 4
-
-    def ocr_extraction(self, document: Document, provider: PdfProvider) -> ProviderPageLines:
-        page_list = [page for page in document.pages if page.text_extraction_method == "surya"]
+    def get_ocr_images_boxes_ids(self, document: Document, pages: List[PageGroup], provider: PdfProvider):
+        highres_images, highres_boxes, line_ids = [], [], []
+        for document_page in pages:
+            page_highres_image = document_page.get_image(highres=True)
+            page_highres_boxes = []
+            page_line_ids = []
+
+            page_size = provider.get_page_bbox(document_page.page_id).size
+            image_size = page_highres_image.size
+            for block in document_page.contained_blocks(document):
+                block_lines = block.contained_blocks(document, [BlockTypes.Line])
+                block_detected_lines = [block_line for block_line in block_lines if block_line.text_extraction_method == 'surya']
+                
+                block.text_extraction_method = 'surya'
+                for line in block_detected_lines:
+                    line_polygon = copy.deepcopy(line.polygon)
+                    page_highres_boxes.append(line_polygon.rescale(page_size, image_size).bbox)
+                    page_line_ids.append(line.id)
+
+            highres_images.append(page_highres_image)
+            highres_boxes.append(page_highres_boxes)
+            line_ids.append(page_line_ids)
+
+        return highres_images, highres_boxes, line_ids
+
+    def ocr_extraction(self, document: Document, pages: List[PageGroup], provider: PdfProvider, images: List[any], line_boxes: List[List[float]], line_ids: List[List[BlockId]]):
+        if sum(len(b) for b in line_boxes)==0:
+            return
 
-        # Remove tables because we re-OCR them later with the table processor
         recognition_results = self.recognition_model(
-            images=[page.get_image(highres=False, remove_tables=not self.enable_table_ocr) for page in page_list],
-            langs=[self.languages] * len(page_list),
-            det_predictor=self.detection_model,
-            detection_batch_size=int(self.get_detection_batch_size()),
+            images=images,
+            bboxes=line_boxes,
+            langs=[self.languages] * len(pages),
             recognition_batch_size=int(self.get_recognition_batch_size()),
-            highres_images=[page.get_image(highres=True, remove_tables=not self.enable_table_ocr) for page in page_list]
+            sort_lines=False
```

Comment:
We're not sorting lines, since we need to map back to the `line_id` that was used previously 

This is fine, but we should probably sort the OCR Lines in the line builder before merging into the page. There's currently no sorting of any kind otherwise for OCRed pages